### PR TITLE
Adapt 687  improve template inheritance

### PR DESCRIPTION
**Pull request by katrialesser**
# READY BUT NEEDS TO BE MERGED AT THE RIGHT TIME
- evaluating option to provide blocks to extend from stanford_basic
- needs to be merged WHEN/after Stanford Basic theme has been updated to use twig blocks, so these changes will work.

# Summary
- Update this subtheme to @extend from stanford_basic better, assuming we add blocks into stanford_basic templates.
- Background on this change in stanford basic is here: https://stanfordits.atlassian.net/browse/D8CORE-2677

# Review By (Date)
- future date

# Urgency
- low, has related accessibility issues it would HELP with

# Steps to Test

1. Pull down the branch for this pr for the saa_victoria_theme
2. Check out this branch of stanford_basic: https://github.com/SU-SWS/stanford_basic/pull/183
3. Clear the site cache
4. See that the page sections render properly. There shouldn't be visual changes. Review code in this pr and in the pr for stanford_basic: https://github.com/SU-SWS/stanford_basic/pull/183

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket ADAPT-687
- this ticket prompted this d8core ticket for updating stanford basic: https://stanfordits.atlassian.net/browse/D8CORE-2677
- Other PRs - this one in stanford basic: https://github.com/SU-SWS/stanford_basic/pull/183 
- This pr needs to be merged **WHEN/after Stanford Basic theme has been updated** (see pr above) to use twig blocks, so these changes will work.
- Once the release of Stanford Basic is ready, then **this branch of the victoria theme needs an updated package.json file with that stanford_basic version as a dependency**. @sherakama started this requirement here: https://github.com/SU-SWS/saa_victoria_subtheme/pull/122/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R10
- cc @cjwest

### CODE REVIEW

File: templates/page.html.twig [hunk 1]
```diff
@@ -1,10 +1,28 @@
-{%- include "@saa_victoria_subtheme/components/brand-bar/saa--brand-bar.html.twig" with { modifier_class : "su-brand-bar--dark" } -%}
 
-{%- if page.help -%}
-  {{ page.help }}
-{%- endif -%}
+{# Need to get this working so it's formatted better, must be missing something:  #}
```
Comment (katrialesser) [inline templates/page.html.twig]: @sherakama I think you're going to very quickly see what I'm doing wrong here with the @extend page template line, but if you have a minute can you help me see how to get this working better having the reference path start from the stanford basic folder?

Comment (katrialesser) [reply templates/page.html.twig]: When I try the line
{%extends "@stanford_basic/templates/page.html.twig"%}
I get the error 
Twig\Error\LoaderError: Template "@stanford_basic/templates/page.html.twig" is not defined in "themes/custom/saa_victoria_subtheme/templates/page.html.twig" at line 3

File: templates/page.html.twig [hunk 2]
```diff
@@ -1,26 +1,27 @@
-{%- include "@saa_victoria_subtheme/components/brand-bar/saa--brand-bar.html.twig" with { modifier_class : "su-brand-bar--dark" } -%}
+{#
+/**
+ * Page.html.twig from stanford_basic.
+ */
+#}
+{%- extends "@stanford_basic/page.html.twig"  -%}
```
Comment (sherakama) [inline templates/page.html.twig]: @katrialesser you don't need to include `templates` in the path of the namespace. That is assumed in most cases.

File: templates/page.html.twig [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
-</header>
+    </section>
+  </header>
+{% endblock %}
 
-<div id="page-content" class="page-content">
-  {{ page.content }}
-</div>
-
-<footer id="footer">
```
Comment (sherakama) [inline templates/page.html.twig:23]: Are you sure you want to remove the `<footer>` tag?

File: templates/page.html.twig [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
+{# Override the footer #}
+{% block block_footer %}
   {%- include "@saa_victoria_subtheme/components/fat-footer/saa-fat-footer.html.twig" with { modifier_class : "test" } -%}
```
Comment (sherakama) [inline templates/page.html.twig]: Also, can we get rid of the `test` class for this?

File: templates/page.html.twig (hunk 2 above)
Comment (katrialesser) [reply templates/page.html.twig]: Gotcha, thank you!

File: templates/page.html.twig (hunk 3 above)
Comment (katrialesser) [reply templates/page.html.twig:23]: Good catch -earlier I was playing with having that <footer> wrapper exist outside the block, I fixed this.

**Review by sherakama — APPROVED** [review]
GTG

**Pull request merged**